lessons/misc/intro-git/lesson.md

```diff
@@ -0,0 +1,394 @@
+downloading `git bash`, which is packaged with [git](https://git-scm.com/) (a
+common version control system).  This contains both Git and a a `bash`
+terminal. To install Git on a Mac, you will need to install XCode from the Mac
+App Store. Warning: XCode is a very large download, and one you'll likely want
+to do well before the lesson!  Alternatively, if you have macOS 10.9
+(Mavericks) or higher, simply open a Terminal instance and run
+
+```
+git --version
+```
+
+to commence the Git installation process. 
+
+Finally, Linux users can install Git using `apt-get` at the command line:
+
+```
+sudo apt-get install git
+```
+
+Finally, you will need to make a free [GitHub][github] account for the latter
+half of this lesson. 
+
+# Why version control?
+
+Version control is a means of tracking changes made to files. These changes are
+tracked alongside detailed metadata on the nature of the change, when it was
+made, and who made it. This is extremely useful, especially when working in
+teams or even for your own records 6 months in the future (because you _will_
+forget things!)
+
+To understand how incredibly powerful version control is, think about these
+questions: How many files of different versions of a manuscript or thesis do
+you have laying around after getting feedback from your supervisor or
+co-authors? Have you ever wanted to experiment with your code or your
+manuscript and need to make a new file so that the original is not touched?
+Have you ever deleted something and wish you hadn't? Have you ever forgotten
+what you were doing on a project? All these problems are fixed by using version
+control (git)!
+
+Today, we are going to cover a basic Git workflow via the command line.  We'll
+be using relatively simple `bash` operations to navigate and work with files
+throughout: if you need a more detailed refresher on these, feel free to refer
+back to our [Intro Bash][bash-intro] material.
+
+Later, as time allows, we'll briefly cover how to bring your Git-controlled
+files to GitHub, a web service that both hosts Git-controlled folders (also
+known as _repositories_, or repos for short) and facilitates collaboration on
+code-based projects.
+
+# Configuring Git
+
+To start off, we have to provide Git with some basic information about ourselves.
+
+First, let's set a username and email:
+
+```bash
+git config --global user.name "Luke Skywalker"
+git config --global user.email "l.skywalker@tattooinet.org"
+```
+
+Git will use `vim` as its default text editor. If you're more inclined to use
+`nano` or `emacs`, the editor can be changed as well:
+
+```bash
+git config --global core.editor nano
+```
+
+# Getting started with Git
+
+## Initializing a repo - `git init`
+
+To use Git, we have to first designate an existing folder as a 'Git
+repository', which will tell Git to start tracking files within it. This will
+create a hidden folder called `.git` in the folder that stores Git-specific
+information.
```

Comment:
Maybe insert: "If you want to see this folder, remember, you can use `ls -a` to list all files and folders in the current working directory, including hidden ones"